### PR TITLE
Controller Documentation

### PR DESCRIPTION
**Pull request by Holzhaus**

### CODE REVIEW

**Comment by Holzhaus** [comment]
I something like this desired? I don't want to fix all this documentation if it's not desired anyway.

**Comment by daschuer** [comment]
I think it is desired. But can we effort the maintenance burden? When I correctly remember it is already hard to keep the wiki updated.

**Comment by Holzhaus** [comment]
> But can we effort the maintenance burden? When I correctly remember it is already hard to keep the wiki updated.

I only added controller that are included in Mixxx. 
I think it makes sense to have documentation for these mappings in the manual as the Wiki page quickly becomes tedious to read when the mapping changes between mixxx versions. We already have some mappings where the wiki page is just a changelog that is completely useless to the end user.

I think adding documentation to the manual opens the possibility to review the documentation properly, too. 

I don't think there's a big maintenance burden, we just need to make sure that anytime a controller mapping is changed the manual page is updated, too.

I suggest the following workflow:
1. Contributor opens PR for a controller mapping
2. We request to add a PR for documentation to the correct branch of this repo
3. We review the documentation and the mapping
4. We merge the mapping only if both the mapping and the documentation are considered ready to merge.

**Comment by Holzhaus** [comment]
Ok, this is ready to review now. Please have a look.

All obvious formatting issues have been fixed. I removed most of the controller images because I doubt we have the copyright owner's permission and they also don't really add value. Schematic images and images with annotations have been kept.

I also removed all programming related info (MIDI value documentation, `lsusb` output, etc.), because that belongs into the wiki. Changelogs, review links, etc. have been removed, too.

**Comment by Be-ing** [comment]
How does https://github.com/mixxxdj/mixxx/wiki/Hardware%20compatibility fit into this? Should this be moved into the manual?

**Comment by Holzhaus** [comment]
> How does https://github.com/mixxxdj/mixxx/wiki/Hardware%20compatibility fit into this? Should this be moved into the manual? I am unsure about that because it lists prices which can change more frequently than we make releases.

I don't think we should put current street prices into the manual. We could use the MSRP or just keep it as simple list (i.e. keep the [current "Controllers" page as-is](https://deploy-preview-122--mixxx-manual.netlify.app/chapters/controllers.html)). The manual is for looking up how your controller works with Mixxx, not for using it as a purchase advice guide.

**Comment by Be-ing** [comment]
We could make https://github.com/mixxxdj/mixxx/wiki/Hardware%20compatibility link to the manual pages for controllers instead of wiki pages.

**Comment by Holzhaus** [comment]
Either that or we keep the wiki pages and replace the mapping description with a link to the manual page. Maybe the latter is preferable because some wiki pages contain useful information for mapping developer that I didn't include here (e.g. MIDI message documentation). Also, old Mixxx version include a link to the wiki page, so maybe we shouldn't remove it.

**Comment by Be-ing** [comment]
Sounds good to me

**Comment by Holzhaus** [comment]
Ok. Shall we merge?

**Comment by Holzhaus** [comment]
Merge?

**Comment by Holzhaus** [comment]
@daschuer Merge?

**Comment by daschuer** [comment]
The wiki link links to the wiki main page. I guess an average user is pretty lost looking for community mappings. 
I think we need to add some words how to search for them and where. 
The standard GitHub Search feature is not very helpful IMHO.

File: source/chapters/controllers/behringer_cmd_mm_1.rst [hunk 1]
```diff
@@ -0,0 +1,229 @@
+Behringer CMD MM-1
+==================
+
+.. figure:: ../../_static/controllers/behringer_cmd_mm_1_overview.png
```
Comment (daschuer) [inline source/chapters/controllers/behringer_cmd_mm_1.rst:4]: This is a 7,2 MB picture loading slooooow

File: source/chapters/controllers/behringer_cmd_studio_4a.rst [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+   :figclass: pretty-figures
+
+   Behringer CMD Studio 4a (top view)
+
+The Behringer CMD Studio 4a is a 2 deck controller that supports 4
+virtual decks and has a built in 4 channel (one stereo master, one
+stereo headphones) USB sound card built in.
+
+-  `Mixxx Forum Thread <http://www.mixxx.org/forums/viewtopic.php?f=7&t=7868>`__
+-  `Manufacturer’s product page <http://www.music-group.com/Categories/Behringer/Computer-Audio/DJ-Controllers/CMD-STUDIO-4a/p/P0809/Features>`__
```
Comment (daschuer) [inline source/chapters/controllers/behringer_cmd_studio_4a.rst:18]: dead

File: source/chapters/controllers/denon_mc6000mk2.rst [hunk 3]
```diff
@@ -0,0 +1,120 @@
+Denon MC6000MK2
+===============
+
+-  `Manufacturer’s product page <http://denondj.com/products/view/mc6000mk2>`__
```
Comment (daschuer) [inline source/chapters/controllers/denon_mc6000mk2.rst:4]: dead

File: source/chapters/controllers/denon_sc2000.rst [hunk 4]
```diff
@@ -0,0 +1,9 @@
+Denon SC2000
+============
+
+-  `Manufacturer’s product page <http://denondj.com/products/view/dn-sc2000>`__
```
Comment (daschuer) [inline source/chapters/controllers/denon_sc2000.rst:4]: dead

File: source/chapters/controllers/gemini_firstmix.rst [hunk 5]
```diff
@@ -0,0 +1,9 @@
+Gemini FirstMix
+===============
+
+-  `Manufacturer’s product page <http://geminisound.com/product/firstmix>`__
```
Comment (daschuer) [inline source/chapters/controllers/gemini_firstmix.rst:4]: dead

File: source/chapters/controllers/hercules_dj_console_4_mx.rst [hunk 6]
```diff
@@ -0,0 +1,319 @@
+Hercules DJ Console 4-Mx
+========================
+
+The Hercules DJ Console 4-Mx is a USB controller with a built in sound card. It is very similar to the [[Hercules DJ Console RMX]], but with switches for 4 decks.
+
+-  `Manufacturer’s product page <http://www.hercules.com/us/DJ-Music/bdd/p/141/dj-console-4-mx/>`__
```
Comment (daschuer) [inline source/chapters/controllers/hercules_dj_console_4_mx.rst:6]: dead

File: source/chapters/controllers/hercules_dj_console_rmx.rst [hunk 7]
```diff
@@ -0,0 +1,258 @@
+Hercules DJ Console RMX
+=======================
+
+-  `Manufacturer’s product page <http://www.hercules.com/us/legacy/bdd/p/62/dj-console-rmx/>`__
```
Comment (daschuer) [inline source/chapters/controllers/hercules_dj_console_rmx.rst:4]: dead

File: source/chapters/controllers/hercules_dj_console_rmx_2.rst [hunk 8]
```diff
@@ -0,0 +1,212 @@
+Hercules DJ Console RMX 2
+=========================
+
+-  `Manufacturer’s product page <http://www.hercules.com/us/DJ-Music/bdd/p/193/djconsole-rmx-2/>`__
```
Comment (daschuer) [inline source/chapters/controllers/hercules_dj_console_rmx_2.rst:4]: dead

File: source/chapters/controllers/hercules_djcontrol_air.rst [hunk 9]
```diff
@@ -0,0 +1,107 @@
...
+
+This controller have built-in soundcard with mixer output (3.5mm jack)
+and headphones output (6.35mm jack).
+
+The main fun and uncommon feature of this controller is its proximity
+sensor (IR) located above EQ potentiometers and between top horizontal
+vu-meter.
+
+-  `Forum Thread <https://www.mixxx.org/forums/viewtopic.php?f=7&t=3263&p=25704&hilit=air#p25704>`__
+-  `Manufacturer’s product page <http://www.hercules.com/fr/legacy/bdd/p/174/djcontrol-air/>`__
```
Comment (daschuer) [inline source/chapters/controllers/hercules_djcontrol_air.rst:20]: dead

File: source/chapters/controllers/hercules_djcontrol_compact.rst [hunk 10]
```diff
@@ -0,0 +1,97 @@
+Hercules DJ Control Compact
+===========================
+
+-  `Manufacturer’s product page <https://www.hercules.com/us/DJ-Music/bdd/p/253/djcontrol-compact/>`__
```
Comment (daschuer) [inline source/chapters/controllers/hercules_djcontrol_compact.rst:4]: dead

File: source/chapters/controllers/hercules_djcontrol_instinct.rst [hunk 11]
```diff
@@ -0,0 +1,95 @@
+Hercules DJ Control Instinct (S)
+================================
+
+This is the cheapest all-in-one controller with an integrated sound
+card. The DJ Control Instinct S is functionally identical to the DJ
+Control Instinct, but has a brushed stainless finish.
+
+-  Manufacturer’s product pages:
+
+   -  `DJ Control Instinct S <http://www.hercules.com/us/leisure-controllers/bdd/p/248/djcontrol-instinct-s-series/>`__
```
Comment (daschuer) [inline source/chapters/controllers/hercules_djcontrol_instinct.rst:10]: dead

File: source/chapters/controllers/hercules_djcontrol_instinct.rst [hunk 12]
```diff
@@ -0,0 +1,95 @@
...
+   -  `DJ Control Instinct <http://www.hercules.com/us/DJ-Music/bdd/p/187/djcontrol-instinct/>`__
```
Comment (daschuer) [inline source/chapters/controllers/hercules_djcontrol_instinct.rst:11]: dead

File: source/chapters/controllers/hercules_p32_dj.rst [hunk 13]
```diff
@@ -0,0 +1,101 @@
+Hercules P32 DJ
+===============
+
+.. figure:: https://www.hercules.com/wp-content/uploads/2018/05/HerculesDJ_HerculesP32DJ_top-1400x747.jpg
+   :alt: https://www.hercules.com/wp-content/uploads/2018/05/HerculesDJ_HerculesP32DJ_top-1400x747.jpg
+
+   https://www.hercules.com/wp-content/uploads/2018/05/HerculesDJ_HerculesP32DJ_top-1400x747.jpg
+
+-  `Manufacturer’s product page <http://www.hercules.com/uk/advanced-controllers/bdd/p/258/hercules-p32-dj/>`__
```
Comment (daschuer) [inline source/chapters/controllers/hercules_p32_dj.rst:9]: dead

File: source/chapters/controllers/ion_discover_dj.rst [hunk 14]
```diff
@@ -0,0 +1,57 @@
+Ion Discover DJ
+===============
+
+The ION Discover DJ is a cheap budget MIDI controller meant to “discover DJ-ing” for people with a potential interest. It does not have a built in sound card, so a `splitter
+cable <hardware%20compatibility#splitter%20cables>`__ or `separate sound card <hardware%20compatibility#USB%20sound%20cards>`__ is required to use headphones with it.
+
+-  `Manufacturer’s product page <http://www.ionaudio.com/products/details/discover-dj>`__
```
Comment (daschuer) [inline source/chapters/controllers/ion_discover_dj.rst:7]: dead

**Review by daschuer — CHANGES_REQUESTED** [review]
Many external links are dead. I have checked only the first.

File: source/chapters/controllers/akai_lpd8.rst [hunk 15]
```diff
@@ -0,0 +1,9 @@
+Akai LPD8
+=========
+
+-  `Manufacturer’s product page <http://www.akaipro.com/product/lpd8>`__
+-  `Forum thread <http://mixxx.org/forums/viewtopic.php?f=7&t=4477>`__
+
+.. note::
+   Unfortunately a detailed description of this controller mapping is still missing.
+   If you own this controller, please consider contributing one.
```
[the remark this answers is not in the document]
Comment (Holzhaus) [reply source/chapters/controllers/akai_lpd8.rst:9]: What contributing guide? Do we have a guide for writing controller mapping documentation?

File: source/chapters/controllers.rst [hunk 16]
```diff
@@ -0,0 +1,16 @@
...
+Controllers
+***********
+
+Mixxx ships with a number built-in :term:`MIDI` or :term:`HID` mappings, so the following controllers are supported out of the box.
+
+Third-party controller mappings that were contributed by the community but are not yet included in Mixxx can be found at Mixxx'
+`Controller Mapping Forum <https://mixxx.discourse.group/c/controller-mappings/10>`__ and the
+`Wiki <https://github.com/mixxxdj/mixxx/wiki>`__.
+
+.. toctree::
```
[the remark this answers is not in the document]
Comment (Holzhaus) [reply source/chapters/controllers.rst:12]: AFAIK that's not possible. Sub-toctrees inherit the numbering from their parent, and I see no way to disable this.

File: source/chapters/controllers/akai_lpd8.rst (hunk 15 above)
[the remark this answers is not in the document]
Comment (daschuer) [reply source/chapters/controllers/akai_lpd8.rst:9]: I think not explicit. Maybe we place two sentences somewhere and link to them here.

File: source/chapters/controllers/reloop_jockey_3_master_edition.rst [hunk 17]
```diff
@@ -0,0 +1,156 @@
...
+control 4 Channels. It has a built-in sound card with 6 channels in and 4
+channels out. It features both balanced 1/4" and unbalanced RCA master outputs
+plus RCA booth outputs with an independent volume control. It can also be used
+as a stand-alone mixer, or used to mix an analog source with music files in
+Mixxx. This device requires a power adapter in addition to the USB cable. This
+device is not USB MIDI class compliant. Its signals are translated to MIDI by
+special drivers on Windows and macOS. *There is no driver available for Linux.*
+
+-  `Mixxx Forum Thread <https://mixxx.discourse.group/t/create-a-mapping-for-reloop-jockey-3-me/13703>`__
+-  `Manufacturer’s product page <http://www.reloop.com/reloop-jockey-3-me>`__
```
Comment (daschuer) [inline source/chapters/controllers/reloop_jockey_3_master_edition.rst:14]: Dead

File: source/chapters/controllers/tascam_us_428.rst [hunk 18]
```diff
@@ -0,0 +1,10 @@
+Tascam US-428
+=============
+
+-  `Manufacturer’s product page <http://www.tascam.com/products/us-428.html>`__
```
Comment (daschuer) [inline source/chapters/controllers/tascam_us_428.rst:4]: Dead

**Review by daschuer — CHANGES_REQUESTED** [review]
Some more broken links

File: source/chapters/controllers/native_instruments_traktor_kontrol_s4_mk2.rst [hunk 19]
```diff
@@ -0,0 +1,113 @@
...
+
+The Kontrol S4 MK2 is a 4 deck all-in-one controller with a sturdy build quality and integrated sound card. The MK2 has substantial improvements over the S4 MK1, including large multicolor buttons.
+The MK1 is not supported and cannot be supported because it uses a proprietary communication protocol exclusive to Traktor. The MK2 uses the standard HID protocol (also used by keyboards & mice) to
+send and receive signals from a computer, so it can work with Mixxx. The easiest way to tell the MK1 apart from the MK2 is the appearance of the jog wheel. On the MK1, the top of the jog wheel is
+black plastic; on the MK2, the top of the jog wheel is shiny aluminum.
+
+The Kontrol S4 Mk2 can run from USB bus power. Using the separate power supply increases the brightness of the LEDs, which is helpful for using it in daylight, and increases the volume of the
+headphone output.
+
+-  `Manufacturer’s product page <https://www.native-instruments.com/en/products/traktor/dj-controllers/traktor-kontrol-s4/>`__
```
[the remark this answers is not in the document]
Comment (Holzhaus) [reply source/chapters/controllers/native_instruments_traktor_kontrol_s4_mk2.rst:12]: No, it doesn't. Maybe there was a bug on the NI webserver?

**Comment by Holzhaus** [comment]
Should we rename the "controllers" section to "hardware"? That way we leave the door open for non-controller hardware, e.g. Soundcards, DVS hardware, etc.

**Comment by Holzhaus** [comment]
I vastly improved the Roland DJ-505 mapping description (didn't bother to do it in the wiki).

**Review by daschuer — APPROVED** [review]
LGTM now. Thank you.

How do we want to proceed?
Delete the wiki pages and redirect to the manual? 

Before, we need to make sure contributors are aware how to improve the controller manual and which part is mandatory when contributing a new mapping.

**Comment by Holzhaus** [comment]
> LGTM now. Thank you.
> 
> How do we want to proceed?
> Delete the wiki pages and redirect to the manual?

Yes.

> Before, we need to make sure contributors are aware how to improve the controller manual and which part is mandatory when contributing a new mapping.

https://github.com/mixxxdj/mixxx/wiki/Contributing-Mappings#documenting-the-mapping

**Comment by daschuer** [comment]
Ok, so we can merge now?

**Comment by Holzhaus** [comment]
Yes.

**Comment by daschuer** [comment]
Ok, thank you.

**Pull request merged**